env/env_posix.cc

```diff
@@ -843,6 +843,15 @@ class PosixEnv : public Env {
 #endif
   }
 
+  virtual uint64_t NowCPUNanos() override {
+#if defined(OS_LINUX) || defined(OS_FREEBSD) || defined(OS_AIX)
+    struct timespec ts;
+    clock_gettime(CLOCK_THREAD_CPUTIME_ID, &ts);
```

Comment:
This looks like a useful API.

Reply:
I wonder if we should handle non-ok return values. Shouldn't hurt.

Reply:
Maybe follow it up earlier. I just copy&paste from NowCPU(), so it's consistent.